Cleanup countries per product

mikkamp opened the pull request:
### Changes proposed in this Pull Request:

This PR is a follow-on to #791

The PR adds the functionality to schedule a particular set of jobs to run for a specific version. These jobs will only be triggered when updating the plugin. We use this code to trigger the job `CleanupProductTargetCountriesJob` which removes any previously synced products for countries other then the main target country.

Since we needed to reuse the code to get the main target country, this was moved to a helper function in the MerchantCenterService (this also removed an added dependency in the BatchProductHelper).

Closes #790 

### Detailed test instructions:

1. Start with trunk and a site which has multiple target countries selected
2. Sync all the products and confirm they appear for all the countries in https://merchants.google.com/
3. Checkout this PR and bump the plugin version to 1.0.1
4. Wait till all scheduled actions have been completed (we should see scheduled actions like `gla/jobs/cleanup_product_target_countries/create_batch`)
5. After waiting a bit confirm that the countries (except main target country) have been removed in https://merchants.google.com/
6. The remaining products should have the shipping set to multiple countries

Test with WR to update `@since` versions (WR 2.0.4 or newer needs to be used):
`wr vb:replace --folder=<folder where you have the plugin installed> --product_version=1.2.3 https://github.com/woocommerce/google-listings-and-ads/tree/trunk`

### Changelog entry
* Fix - Cleanup target countries per product.

mikkamp committed:
Add versions when triggering install functions

mikkamp committed:
Run jobs on plugin update

mikkamp committed:
Helper function for retrieving main target country

mikkamp committed:
Job for cleaning up product target countries

nima-karimi reviewed (commented):
The code looks good and it deletes all of the previously synced products on update 💯 But it doesn't re-sync the deleted products so that they have the right shipping set. We could add a call to update in there too maybe? Or run the update job separately for those products if that's possible.

nima-karimi commented on src/Jobs/Update/CleanupProductTargetCountriesJob.php:
```diff
@@ -0,0 +1,55 @@
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Jobs\Update
+ */
+class CleanupProductTargetCountriesJob extends AbstractProductSyncerBatchedJob {
```
I think it's now time to add the `@since` annotation to all our newly added classes and functions. Same for the `PluginUpdate` job and the other methods added in this PR.

mikkamp commented:
> But it doesn't re-sync the deleted products so that they have the right shipping set.

I was initially relying on your PR to handle that part of the process. But forgot that it isn't triggered automatically. The code I wrote can trigger multiple Jobs on update, so let me see if I can just add the Update one to the list.

mikkamp committed:
Trigger UpdateAllProducts to update shipping

mikkamp committed:
Add since for new functions and classes

mikkamp committed:
Entry for woorelease to autoreplace since version

mikkamp commented:
@nima-karimi It's ready for another review. I added the UpdateAllProducts job so it will update all shipping countries.
I added the since variables and included a test (in the PR description) to make sure they get replaced with WR.

nima-karimi reviewed (approved):
All working for me. Thanks, @mikkamp. I just tested it again and had all my products deleted and resubmitted with new shipping. The `@since` doc annotation is also replaced when running the `woorelease` command.

One minor thing: Do we also need to add the `@since` to all of the methods of a new class or is it enough to just add it to the class itself?

mikkamp commented:
Thanks for the review. I think it's cleaner to add it only to the class, then as new functions get added we can add since to just those functions. Otherwise we'd almost need a since in every function. But that's personal preference, so open to change.
